src/platform/test.rs

```diff
@@ -414,6 +414,54 @@ fn receiver_set() {
 }
 
 #[test]
+#[cfg(not(any(feature = "force-inprocess", target_os = "windows", target_os = "android")))]
+fn receiver_set_eintr() {
+    let (tx0, rx0) = platform::channel().unwrap();
+    let (tx1, rx1) = platform::channel().unwrap();
+    let mut rx_set = OsIpcReceiverSet::new().unwrap();
+    rx_set.add(rx1).unwrap();
+    let child_pid = unsafe {
+        fork(|| {
+            // Let the parent know we're ready
+            tx0.send(b"\x00", vec![], vec![]).unwrap();
+            // Send the result of the select back to the parent
+            match rx_set.select() {
+                Ok(result) => {
+                    for evt in result {
+                        if let OsIpcSelectionResult::DataReceived(_, data, _, _) = evt {
+                            assert!(data == b"\x00");
```

Comment:
Well, I see your point  -- but if you really want to go down this road, you'd have to match *all* the result values manually in the child process, rather than using any asserts/unwraps etc. Or maybe you could use something with `set_hook()` and/or `catch_unwind()`, which would allow sticking with the usual assert/unwrap stuff -- but that seems tricky to get right as well.

To be honest, I'm not sure either is worth the effort. Just send back the correct result I'd say, and panic the child process otherwise -- like the other multi-threaded/multi-process tests do. Running into a test failure is not the common case; when someone does, they  kinda just have to figure out that they need to run the tests with `-- --nocapture` to see the additional error output... Not ideal, but good enough for the tests.